guide_drafts/model-derives.md

```diff
@@ -0,0 +1,769 @@
+             <(A,) as diesel::types::FromSqlRow<(SA,), DB>>
+   = note: required because of the requirements on the impl of `diesel::Queryable<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` for `diesel_demo_cli::models::User`
+   = note: required because of the requirements on the impl of `diesel::LoadQuery<_, diesel_demo_cli::models::User>` for `diesel_demo_cli::schema::users::table`
+```
+
+Notice the compiler is indicating a trait is not implmented for using the `.load()` method. FromSqlRow is expecting a tuple that looks like `(i32, std::string::String, std::string::String, std::string::String)`, but we currently only have a tuple consisting of `(i32,)`.
+
+This trait bound is illustrated below in the "hand-written" derive example.
+
+```rust
+
+// Implementing the Queryable trait by hand
+
+use diesel::query_source::Queryable;
+use diesel::backend::Backend;
+use diesel::types::{Integer, Text, HasSqlType, FromSqlRow};
+
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: String,
+}
+
+impl<ST, DB> Queryable<ST, DB> for User where
+    DB: Backend + HasSqlType<ST>,
+    (i32, String, String, String): FromSqlRow<ST, DB> {
+    
+    type Row = (i32, String, String, String);
+
+    fn build(row: &Self::Row) -> Self {
+        User {
+            id: row.0,
+            first_name: row.1,
+            last_name: row.2,
+            email: row.3,
+        }
+    }
+}
+
+
+```
+
+The advantages of using the derive annotation should be clear here, as the more columns you have, the more code you would have to write. You can thank Rust for this zero-cost-abstraction!
+
+## <a name="insertable"></a>Insertable
+
+Using the [#[derive(Insertable)]](http://docs.diesel.rs/diesel/prelude/trait.Insertable.html) trait annotation on your model struct allows records to be inserted into the database via that struct. You will need `Insertable` when using the `diesel::insert(&my_new_row).into(my_table::table)` API. You may pass a reference to a single struct, `Vec` or slice to insert a single record or multiple records at once.
+
+When implementing `Insertable`, you probably won't be setting the auto-incremented `Id` field of the row. For this reason, it's not advisable to use `Queryable` and `Insertable` on the same struct due to the field number constraints of `Queryable`. Create another struct that you may use for database insertions that will have all the fields you would like to set. This section will not cover nullable fields (we'll cover that in [AsChangeset](#aschangeset)), so we will assume every field must have data in our example. When making a separate struct for database inserts, Diesel needs to know the corresponding table name, so the struct must also be annotated with the `#[table_name="some_table_name"]` attribute. If your new struct has different field names, each of them may be annotated with `#[column_name(some_column_name)]`.
```

Comment:
Other than `id`, timestamp fields like `created_at` and `updated_at` are also usually omitted in the Insertable structs